arch/arm/core/cortex_a_r/Kconfig

```diff
@@ -111,6 +111,24 @@ config CPU_CORTEX_R52
 	help
 	  This option signifies the use of a Cortex-R52 CPU
 
+config CPU_CORTEX_R52_ICACHE_FLASH_WAY
+	int "I-Cache Flash way"
+	depends on CPU_CORTEX_R52
+	range 0 4
+	default 0
+	help
+	  Configure I-Cache ways for Flash interface. Default is reset value, all
+	  I-Cache ways are allocated for AXIM interface.
+
+config CPU_CORTEX_R52_DCACHE_FLASH_WAY
+	int "D-Cache Flash way"
+	depends on CPU_CORTEX_R52
+	range 0 4
```

Comment:
If I understand correctly, this would allow configuring 4 cache ways to flash and 4 cache ways to AXIM, which is not correct. There are 4 cache ways and they need to be split between both flash and AXIM, we cannot allow any combination.

Also because this is hw configuration, it would make more sense to me to have it as a dt prop enum, where user can select between flash/axim: `0/4`, `1/3`, `2/2`, `3/1`, `4/0`. This would solve above problem as well.

Reply:
No, the Kconfigs are meant cache ways for Flash interface. So for example if `4` is chosen (way 0-3 allocated for Flash), this also means no cache way is allocated for AXIM:

![image](https://github.com/user-attachments/assets/3da4001a-8843-46e4-b8a3-45487398f245)

Reply:
ok, I overlooked the kconfigs, it makes more sense now. I still think this should be better a dt prop than a kconfig option, but is not a request for change from my side.